### PR TITLE
Dogfooding data accessor

### PR DESCRIPTION
**Pull request by publixsubfan**
Replaces boilerplate in parmetis/repartitioning components with data accessor usage.

### CODE REVIEW

**Comment by rhornung67** [comment]
Is there an award for best PR title? 😄

File: src/libs/blueprint/conduit_blueprint_mesh_partition.cpp [hunk 1]
```diff
@@ -4267,78 +4234,28 @@ node_value_compare_impl2(const LhsDataArray &lhs, const RhsDataArray &rhs, doubl
...
 static bool
-node_value_compare_impl(const Node &lhs, const RhsDataArray &rhs,  double epsilon)
+node_value_compare_flt(const Node &lhs, const Node &rhs, double epsilon)
 {
-    const auto id = lhs.dtype().id();
-    bool retval = true;
-    switch(id)
-    {
-    case conduit::DataType::INT8_ID:
+    double_accessor lhs_data = lhs.as_double_accessor();
```
Comment (cyrush) [inline src/libs/blueprint/conduit_blueprint_mesh_partition.cpp]: minor suggestion: use` float64_accessor`, instead of `double_accessor`

**Comment by cyrush** [comment]
I like what I see -- this simplifies things quite a bit!

DataAccessor helping as we thought. 

Also +1 for the title. 

Are there any suggestions you have that would impact the API of the Accessor?

We can merge the DataAccessor branch soon -- I was planning to a some basic docs about them.

Thanks for looking at this!

File: src/libs/conduit/conduit_node.cpp [hunk 2]
```diff
@@ -15022,6 +15022,13 @@ Node::as_float64_accessor() const
     return float64_accessor(m_data,dtype());
 }
 
+//---------------------------------------------------------------------------//
+index_t_accessor
+Node::as_index_t_accessor() const
```
Comment (cyrush) [inline src/libs/conduit/conduit_node.cpp:15027]: +1

File: src/libs/blueprint/conduit_blueprint_mesh_partition.cpp [hunk 3]
```diff
@@ -4095,73 +4095,26 @@ template<typename OutDataArray>
...
-    }
-    case conduit::DataType::UINT64_ID:
+    if (idt.is_integer())
     {
-        DataArray<uint64> da = in.value();
-        retval = copy_node_data_impl2(da, out, offset);
-        break;
+        if (idt.id() == conduit::DataType::UINT64_ID)
+        {
+            retval = copy_node_data_impl2(in.as_uint64_accessor(), out, offset);
```
Comment (cyrush) [inline src/libs/blueprint/conduit_blueprint_mesh_partition.cpp]: should we use `is_signed_integer()` and `is_unsigned_integer()`  for this case?

**Review by cyrush — APPROVED** [review]
looks good, added a comment with one question

File: src/libs/blueprint/conduit_blueprint_mesh_partition.cpp (hunk 3 above)
Comment (publixsubfan) [reply src/libs/blueprint/conduit_blueprint_mesh_partition.cpp]: probably - `int64` should correctly copy everything but `uint64`, but splitting up by signed/unsigned is probably clearer

Comment (cyrush) [reply src/libs/blueprint/conduit_blueprint_mesh_partition.cpp]: got it, It wasn't obvious to me -- a comment would have also been fine, but thanks for making the adjustments

**Pull request merged**